exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/ElasticsearchExporterConfigurationIT.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.exporter;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.awaitility.Awaitility.await;
+
+import io.camunda.zeebe.exporter.api.ExporterException;
+import org.junit.Test;
+
+public class ElasticsearchExporterConfigurationIT
+    extends AbstractElasticsearchExporterIntegrationTestCase {
+
+  @Test
+  public void shouldPropagateNumberOfShardsAndReplicas() {
+    // given
+    elastic.start();
+
+    configuration = getDefaultConfiguration();
+
+    // change number of shards and replicas
+    configuration.index.setNumberOfShards(5);
+    configuration.index.setNumberOfReplicas(4);
+
+    esClient = createElasticsearchClient(configuration);
+
+    // when
+    exporterBrokerRule.configure("es", ElasticsearchExporter.class, configuration);
+    exporterBrokerRule.start();
+    exporterBrokerRule.publishMessage("message", "123");
+
+    // then
+    await("index templates need to be created").untilAsserted(this::assertIndexSettings);
```

Comment:
:wrench: Personally, I am not a fan of this construct. Mostly because if it fails it will fail like this: 
`org.awaitility.core.ConditionTimeoutException: Condition with alias 'index templates need to be created' didn't complete within 10 seconds because assertion condition defined as a lambda expression in io.camunda.zeebe.exporter.ElasticsearchExporterConfigurationIT fail on purpose.` (where "fail in purpose" is the failure message I introduced artifically)

So I would prefer a construct:
```
await("indexes are created).untilAsserted(this:indexesGenerated)
assertIndexSettings();
```

Looking at the code which is already there, it's a bit of a mix. So feel free to follow your instinct

Reply:
Changed it a bit.